addons/website_google_map/controllers/main.py

```diff
@@ -24,14 +24,23 @@ class GoogleMap(http.Controller):
 
     @http.route(['/google_map'], type='http', auth="public", website=True, sitemap=False)
     def google_map(self, *arg, **post):
+        PartnerSudo = request.env['res.partner'].sudo()
         clean_ids = []
-        for partner_id in post.get('partner_ids', "").split(","):
-            try:
-                clean_ids.append(int(partner_id))
-            except ValueError:
-                pass
-        partners = request.env['res.partner'].sudo().search([("id", "in", clean_ids),
-                                                             ('website_published', '=', True), ('is_company', '=', True)])
+        domain = [('website_published', '=', True)]
+        if post.get('partner_ids'):
+            for partner_id in post['partner_ids'].split(","):
+                try:
+                    clean_ids.append(int(partner_id))
+                except ValueError:
+                    pass
+            domain += [("id", "in", clean_ids), ('is_company', '=', True)]
+        elif post.get('fn') and hasattr(self, f"_google_map_domain_{post['fn']}"):
+            domain += getattr(self, f"_google_map_domain_{post['fn']}")(**post)
+        else:
+            domain += [(0, '=', 1)]
```

Comment:
Nitpicking but would not it better to just not do the search in that case? Could have:
```python
domain = None
if machin1:
    domain = ...
elif machin2:
    domain = ...

partners = PartnerSudo
if domain is not None:
    partners = PartnerSudo.search(domain + [('website_published', '=', True)])


```

Reply:
`(0, '=', 1)` is a falsy domain, ORM is smart and don't execute anything with it, it just return an empty browseRecord 

```py
In [1]: env['res.partner'].search([('name', '=', 'ok'), (0, '=', 1)])
Out[1]: res.partner()

In [2]: env['res.partner'].search([('name', '=', 'ok')])
2024-05-14 15:13:54,657 816919 DEBUG openerp odoo.sql_db: [50.663 ms] query: SELECT "res_partner"."id" FROM "res_partner" WHERE (("res_partner"."active" = true) AND ("res_partner"."name" = 'ok')) ORDER BY "res_partner"."complete_name" ASC , "res_partner"."id" DESC  

```

Reply:
I was nitpicking to avoid the `(0, '=', 1)` not increasing perf :wink: You would be the only one that does `+ [(0, '=', 1)]`, but as you wish :+1: